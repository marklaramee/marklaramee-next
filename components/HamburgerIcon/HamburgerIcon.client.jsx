import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './styles.module.css';

const HamburgerIcon = ({isOpen}) => (
  <div className={classNames(styles.wrapper, {[styles.open]: isOpen})}>
    <div className={styles.topBlock} />
    <div className={styles.middleBlock} />
    <div className={styles.bottomBlock} />
  </div>
);

HamburgerIcon.propTypes = {
  isOpen: PropTypes.bool.isRequired,
};

export default HamburgerIcon;
