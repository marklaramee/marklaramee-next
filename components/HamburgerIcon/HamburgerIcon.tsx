import styles from './styles/HamburgerIcon.module.css';
import classnames from 'classnames';

export interface props {
    isOpen: boolean; 
}

const HamburgerIcon = ({isOpen}: props) => (
  <button className={classnames(styles.reset, styles.hamburgerButton)}>
    <div className={classnames(styles.wrapper, {[styles.open]: isOpen})}>
      <div className={styles.topBlock} />
      <div className={styles.middleBlock} />
      <div className={styles.bottomBlock} />
    </div>
  </button>
);

export default HamburgerIcon;
