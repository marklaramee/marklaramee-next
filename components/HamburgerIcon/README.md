# HamburgerButton

This component is an hamburger button.

## Sample Usage

```js
import HamburgerButton from 'components/HamburgerButton';

() => <HamburgerButton />;
```

The example above renders a hamburger button.

The `isOpen` prop will determine the visual state of the button (hamburger when `isOpen` is false, and an `X` when `isOpen` is true):

```js
import HamburgerButton from 'components/HamburgerButton';

() => <HamburgerButton isOpen={navigationIsOpen} />;
```

Any props you pass it will be applied to the underlying `button` tag:

```js
import HamburgerButton from 'components/HamburgerButton';

() => <HamburgerButton onClick={doSomething} aria-label="Toggle navigation" />;
```
