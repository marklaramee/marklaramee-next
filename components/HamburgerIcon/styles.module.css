.wrapper {
  display: flex;
  flex-direction: column;
}

.middleBlock {
  display: block;
  height: 2px;
  width: 22px;
  background-color: black;
  position: relative;
  transition: all 250ms;
}

.topBlock {
  content: '';
  position: relative;
  top: calc(var(--spacing-px) * -1);
  width: 22px;
  height: 2px;
  background-color: black;
  transition: all 250ms;
}

.bottomBlock {
  content: '';
  position: relative;
  top: var(--spacing-px);
  left: 0;
  width: 22px;
  height: 2px;
  background-color: black;
  transition: all 250ms;
}

.open .middleBlock {
  background-color: transparent;
}

.open .topBlock {
  transform: rotate(45deg);
  top: 2px;
}

.open .bottomBlock {
  transform: rotate(-45deg);
  top: -2px;
}
