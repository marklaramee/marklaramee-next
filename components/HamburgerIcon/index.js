import HamburgerIcon from './HamburgerIcon.client';

export default HamburgerIcon;
