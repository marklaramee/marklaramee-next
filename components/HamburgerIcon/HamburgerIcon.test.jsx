import {expect, describe, beforeEach, it} from 'vitest';
import renderer from 'react-test-renderer';

import HamburgerIcon from './HamburgerIcon.client';

describe('Component:HamburgerIcon', () => {
  let subject;

  describe(':snapshots', () => {
    describe('when isOpen = true', () => {
      beforeEach(() => {
        subject = renderer.create(<HamburgerIcon isOpen />).toJSON();
      });

      it('renders snapshot', () => {
        expect(subject).toMatchSnapshot();
      });
    });

    describe('when isOpen = false', () => {
      beforeEach(() => {
        subject = renderer.create(<HamburgerIcon isOpen={false} />).toJSON();
      });

      it('renders snapshot', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});
