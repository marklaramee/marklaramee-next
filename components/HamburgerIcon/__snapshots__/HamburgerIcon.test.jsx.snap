// Vitest Snapshot v1

exports[`Component:HamburgerIcon > :snapshots > when isOpen = false > renders snapshot 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "._wrapper_ {
  display: flex;
  flex-direction: column;
}

._middleBlock_ {
  display: block;
  height: 2px;
  width: 22px;
  background-color: black;
  position: relative;
  transition: all 250ms;
}

._topBlock_ {
  content: '';
  position: relative;
  top: calc(var(--spacing-px) * -1);
  width: 22px;
  height: 2px;
  background-color: black;
  transition: all 250ms;
}

._bottomBlock_ {
  content: '';
  position: relative;
  top: var(--spacing-px);
  left: 0;
  width: 22px;
  height: 2px;
  background-color: black;
  transition: all 250ms;
}

._open_ ._middleBlock_ {
  background-color: transparent;
}

._open_ ._topBlock_ {
  transform: rotate(45deg);
  top: 2px;
}

._open_ ._bottomBlock_ {
  transform: rotate(-45deg);
  top: -2px;
}
",
      }
    }
  />,
  <div
    className="_wrapper_"
  >
    <div
      className="_topBlock_"
    />
    <div
      className="_middleBlock_"
    />
    <div
      className="_bottomBlock_"
    />
  </div>,
]
`;

exports[`Component:HamburgerIcon > :snapshots > when isOpen = true > renders snapshot 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "._wrapper_ {
  display: flex;
  flex-direction: column;
}

._middleBlock_ {
  display: block;
  height: 2px;
  width: 22px;
  background-color: black;
  position: relative;
  transition: all 250ms;
}

._topBlock_ {
  content: '';
  position: relative;
  top: calc(var(--spacing-px) * -1);
  width: 22px;
  height: 2px;
  background-color: black;
  transition: all 250ms;
}

._bottomBlock_ {
  content: '';
  position: relative;
  top: var(--spacing-px);
  left: 0;
  width: 22px;
  height: 2px;
  background-color: black;
  transition: all 250ms;
}

._open_ ._middleBlock_ {
  background-color: transparent;
}

._open_ ._topBlock_ {
  transform: rotate(45deg);
  top: 2px;
}

._open_ ._bottomBlock_ {
  transform: rotate(-45deg);
  top: -2px;
}
",
      }
    }
  />,
  <div
    className="_wrapper_ _open_"
  >
    <div
      className="_topBlock_"
    />
    <div
      className="_middleBlock_"
    />
    <div
      className="_bottomBlock_"
    />
  </div>,
]
`;
